src/tools/ScenarioMeasurement/Startup/ProcessTimeParser.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session;
+using Reporting;
+using System;
+using System.Collections.Generic;
+
+
+namespace ScenarioMeasurement
+{
+    internal class ProcessTimeParser : IParser
+    {
+        public void EnableKernelProvider(TraceEventSession kernel)
+        {
+            kernel.EnableKernelProvider((KernelTraceEventParser.Keywords.Process | KernelTraceEventParser.Keywords.Thread | KernelTraceEventParser.Keywords.ContextSwitch));
+        }
+
+        public void EnableUserProviders(TraceEventSession user)
+        {
+            user.EnableProvider("PerfLabProcessTimeEventSource");
```

Comment:
You should not need to enable a user provider here because the events that you use below are only kernel events.

Reply:
This is what I thought. However it would break merging files in Startup.cs because no user trace file will be created. If necessary we could change the conditions of merging files?

Reply:
As discussed put a check in Startup.cs for the existence of the user trace, so we don't have to turn on a fake user trace here.